### PR TITLE
Compress path in title bar by un-expanding tilde

### PR DESCRIPTION
**Pull request by Carreau**
we **might** want to actually create the untildify logic ourself that is probably relatively simple, I just don't  node stdlib enough.

### CODE REVIEW

**Comment by rgbkrk** [comment]
This looks like a good idea to me. I'm not sure how/why this failed the functional tests, so I re-kicked it.

**Comment by jdfreder** [comment]
Looks like the tildify package may be having trouble on the version of electron that we are testing against ??

**Comment by jdfreder** [comment]
Test fails locally too

**Comment by jdfreder** [comment]
Test pass locally on master though...  So yeah, must be the tildify package..

**Comment by rgbkrk** [comment]
We do have our own home directory function (that is used within jupyter-paths). Coupling that with this function

```
function tildify(home, p) {
  const s = path.normalize(p) + path.sep;
  return (s.indexOf(home) === 0 ? s.replace(home + path.sep, '~' + path.sep) : s).slice(0, -1);
}
```

and we're well on our way.

**Comment by rgbkrk** [comment]
Weird, doesn't pass locally for me either. I wonder if it's the synchronous home directory call that `tildify` makes.

**Comment by Carreau** [comment]
I'll update not to use tildify.

**Comment by Carreau** [comment]
Updated.

**Comment by rgbkrk** [comment]
Awesome! I'm going to merge and then make some notes inline.

File: src/notebook/native-window/index.js [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
+  const s = path.normalize(p) + path.sep;
+  return (s.indexOf(home) === 0 ? s.replace(home + path.sep, '~' + path.sep) : s).slice(0, -1);
+}
+
+function tildify(path) {
+  if(!path){
+    return ''
+  }
+  return _tildify(HOME, path)
+}
```
Comment (rgbkrk) [inline src/notebook/native-window/index.js:17]: Since the function is not exported / not given as a public API, we don't need to use a leading underscore.

**Pull request merged**